operator/pkg/resources/statefulset_update.go

```diff
@@ -452,7 +455,15 @@ func (r *StatefulSetResource) podEviction(ctx context.Context, pod, artificialPo
 		return err
 	}
 
-	if patchResult.IsEmpty() {
+	var updateForConfigRestart bool
+	if cond := utils.FindStatusPodCondition(pod.Status.Conditions, ClusterUpdatePodCondition); cond != nil && cond.Reason == ClusterUpdateReasonConfig {
+		// It's okay to remove this condition - it will be reapplied if the cluster remains in "Needs Restart"
+		log.Info("Pod marked as requiring an update for restart-only configuration. Deleting pod",
+			"pod-name", pod.Name)
+		updateForConfigRestart = true
+	}
+
+	if patchResult.IsEmpty() && !updateForConfigRestart {
```

Comment:
Would you mind adding a log line here as well? Hopefully we won't hit it but it will very likely prevent a bit of hair pulling if we do find ourselves asking "why isn't the operator restarting my Pods?" again.

Reply:
Yeah.